api/app/jobs/rdps_sfms.py

```diff
@@ -0,0 +1,177 @@
+                                os.remove(downloaded)
+            except Exception as exception:
+                self.exception_count += 1
+                # We catch and log exceptions, but keep trying to download.
+                # We intentionally catch a broad exception, as we want to try and download as much
+                # as we can.
+                logger.error("unexpected exception processing %s", url, exc_info=exception)
+
+    async def process_model_run(self, model_run_hour: int):
+        """Process a particular RDPS model run"""
+        logger.info(f"Processing RDPS model run {model_run_hour}Z")
+        for key, value in GRIB_LAYERS.items():
+            urls = list(get_regional_model_run_download_urls(self.now, model_run_hour, [value], MAX_MODEL_RUN_HOUR))
+            await self.process_model_run_urls(model_run_hour, key, urls)
+
+    async def process(self):
+        """Entry point for downloading and processing RDPS weather model grib files"""
+        for hour in get_model_run_hours_to_process():
+            try:
+                await self.process_model_run(hour)
+                if self.files_downloaded > 0:
+                    create_model_run_for_sfms(self.session, ModelEnum.RDPS, self.now, hour)
+            except Exception as exception:
+                # We catch and log exceptions, but keep trying to process.
+                # We intentionally catch a broad exception, as we want to try to process as much as we can.
+                self.exception_count += 1
+                logger.error("unexpected exception processing %s model run %d", self.model_type, hour, exc_info=exception)
+
+
+class RdpsJob:
+    async def run(self):
+        # Add some check based on current time and previous runs to determine if we need to proceed
+        logger.info("Begin download and storage of RDPS gribs.")
+
+        # grab the start time.
+        start_time = datetime.now()
+        with get_write_session_scope() as session:
+            rdps_grib = RdpsGrib(session)
+            await rdps_grib.process()
+
+        # calculate the execution time.
+        execution_time = datetime.now() - start_time
+        hours, remainder = divmod(execution_time.seconds, 3600)
+        minutes, seconds = divmod(remainder, 60)
+
+        # log some info.
+        logger.info(
+            "%d downloaded, time taken %d hours, %d minutes, %d seconds (%s)",
+            rdps_grib.files_downloaded,
+            hours,
+            minutes,
+            seconds,
+            execution_time,
+        )
+        # check if we encountered any exceptions.
+        if rdps_grib.exception_count > 0:
+            # if there were any exceptions, return a non-zero status.
+            raise CompletedWithSomeExceptions()
+
+
+def main():
+    """Kicks off asynchronous downloading and storage of RDPS weather model data."""
+    try:
+        logger.debug("Begin download and storage of RDPS NWM data.")
+
+        bot = RdpsJob()
```

Comment:
Minor: I know this is a holdover from other jobs but maybe we can change `bot` to `job`